pkg/api/images.go

```diff
@@ -10,39 +9,43 @@ import (
 
 var performerBox *packr.Box
 var performerBoxMale *packr.Box
+var performerBoxCustom *packr.Box
 
 func initialiseImages() {
 	performerBox = packr.New("Performer Box", "../../static/performer")
 	performerBoxMale = packr.New("Male Performer Box", "../../static/performer_male")
 }
 
-func getRandomPerformerImage(gender string) ([]byte, error) {
+func getRandomPerformerImageUsingName(name, gender, customPath string) ([]byte, error) {
 	var box *packr.Box
-	switch strings.ToUpper(gender) {
-	case "FEMALE":
-		box = performerBox
-	case "MALE":
-		box = performerBoxMale
-	default:
-		box = performerBox
 
+	// If we have a custom path, we should return a new box in the given path.
+	if customPath != "" {
+		if performerBoxCustom != nil {
+			box = performerBoxCustom
+		} else {
+			// We need to set performerBoxCustom at runtime, as this is a custom path, and store it in a pointer.
+			newBox := packr.New("Custom Performer Box", customPath)
+
+			// only grab from this box if it is not empty
+			if len(newBox.List()) > 0 {
+				performerBoxCustom = newBox
+				box = newBox
+			}
+		}
 	}
```

Comment:
Shouldnt this be modified and included  in the initialise images function?
I am getting a lot of memory/cpu usage and the custom images not loading which i am guessing is due to boxes being created for every single randomperformer query
Having sorted my performers by recently added and 30+ performers not having images (so random is used) i am getting huge memory usage/leaks when the custom path is not null (be it an existing/correct path or a non exiting one)